Software/Development/SemesterProject/System/scheduler.c

```diff
@@ -16,28 +25,31 @@ taskControlBlock taskList[maxNumberTasks];
 
 void taskListInitialization(void)
 {
-    int taskListIndex = 0;
-    taskList[taskListIndex].functionPointer = task50us;
-    taskList[taskListIndex].taskState = INACTIVE;
-    taskList[taskListIndex].cyclicity = 50;
-    taskListIndex++;
+    Uint16 taskListIndex = 0;
 
     taskList[taskListIndex].functionPointer = task20ms;
     taskList[taskListIndex].taskState = READY;
-    taskList[taskListIndex].cyclicity = 20000;
+    taskList[taskListIndex].cyclicity = 20000;      // Time in us
+    taskList[taskListIndex].timeLeft = 20000;       // Initialize to cyclicity
     taskListIndex++;
 
-    taskList[taskListIndex].functionPointer = NULL;
+    taskList[taskListIndex].functionPointer = task50us;
+    taskList[taskListIndex].taskState = INACTIVE;
+    taskList[taskListIndex].cyclicity = 50;
+    taskList[taskListIndex].timeLeft = 50;
+    taskListIndex++;
+
+    taskList[taskListIndex].functionPointer = NULL; // End of list
```

Comment:
I'll think about it.

Reply:
I've been analysing what are other options for implementing this and it could be done by performing the commented function into a temporary function and finally getting the value of the counter into a quasi global variable and then defining the length of the array with that counter and finally copying the values of the temporary structure into the final array. However I don't think that's worth it.  The number of tasks might change, but probably not more than a couple of times and there's no worry about using a few bytes more for that struct. I think the main concern to change this was to increase development speed and decrease error probability in future developments but I think is not worth.